pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,450 @@
+    ASK_YES_NO_SAVE: 3,
+    ASK_PASSWORD: 4,
+});
+
+const ProblemState = Object.freeze({
+    REPORTABLE: 0,
+    REPORTING: 1,
+    REPORTED: 2,
+    UNREPORTABLE: 3,
+});
+
+// Avoiding using the proxy to circumvent the GLib cache.
+function get_problem_property(problem, property) {
+    return cockpit
+            .dbus("org.freedesktop.problems", { superuser: "try" })
+            .call(problem.path,
+                  "org.freedesktop.DBus.Properties",
+                  "Get",
+                  ["org.freedesktop.Problems2.Entry", property]);
+}
+
+class FAFWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            problemState: ProblemState.REPORTABLE,
+            process: null,
+            reportLinks: [],
+            message: "",
+        };
+
+        this.onCancelButtonClick = this.onCancelButtonClick.bind(this);
+        this.onReportButtonClick = this.onReportButtonClick.bind(this);
+        this.updateStatusFromBus = this.updateStatusFromBus.bind(this);
+
+        this.updateStatusFromBus();
+    }
+
+    updateStatusFromBus() {
+        get_problem_property(this.props.problem, "Reports")
+                .done((reports) => {
+                    const reportLinks = [];
+                    let reported = false;
+
+                    for (const report of reports[0].v) {
+                        if (report[0] === "ABRT Server") {
+                            if ("URL" in report[1]) {
+                                reportLinks.push(report[1].URL.v.v);
+                            }
+                            reported = true;
+                        }
+                    }
+
+                    this.setState({
+                        problemState: reported ? ProblemState.REPORTED : this.state.problemState,
+                        reportLinks: reportLinks,
+                    });
+                });
+        get_problem_property(this.props.problem, "CanBeReported")
+                .done((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+
+    onCancelButtonClick(event) {
+        this.state.process.close("canceled");
+    }
+
+    onReportButtonClick(event) {
+        if (this.state.problemState === ProblemState.REPORTING) {
+            return;
+        }
+
+        const process = cockpit.spawn(["reporter-ureport", "-d", this.props.problem.ID],
+                                      {
+                                          err: "out",
+                                          superuser: "true",
+                                      })
+                .always(() => {
+                    this.updateStatusFromBus();
+                })
+                .done(() => {
+                    this.setState({ problemState: ProblemState.REPORTED, });
+                })
+                .fail((ex) => {
```

Comment:
Please at least `console.warn()` the exception, so that we have a chance to debug problems with the reporting.

Reply:
It’s not really an exception, though: https://cockpit-project.org/guide/latest/cockpit-spawn.html#cockpit-spawn-fail

Reply:
I can add some output for when it’s killed or exits with a non-zero code, but the latter should be more or less reflected in the status column with a line from stderr.